### PR TITLE
Script to run sanity test on LSI3008 adapter

### PR DESCRIPTION
**Pull request by vrbagalkote**
This script will run few basic test on the adapter

Signed-off-by: Venkat R B <vrbagal1@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by vrbagalkote** [comment]
[job.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/597108/job.txt)

Attached is job.log file.

Thanks,
Venkat.

File: io/disk/lsi3008.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+        """
+        Lists all available Avago adapters (does not need ctlr #>
+        """
+
+        self.controller = int(self.params.get('controller', default='0'))
+        self.disk = self.params.get('disk', default='').strip(" ").split(" ")
+        self.drives = []
+        for i in self.disk:
+            self.drives.append(i)
```
Comment (narasimhan-v) [inline io/disk/lsi3008.py]: `self.disk` and `self.drives` are same. Please use any one.

File: io/disk/lsi3008.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        if process.system(cmd, ignore_status=True, shell=True) != 0:
+            self.fail("Failed to display details of drives and VR vloumes")
+
+    def test_createraid(self):
+
+        """
+        This function creates raid1 array
+        """
+
+        if self.number_of_disk >= 2:
```
Comment (narasimhan-v) [inline io/disk/lsi3008.py]: This test will simply `PASS` when disks are less than 2. Handle that please

File: io/disk/lsi3008.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    def test_hotspare(self):
+
+        """
+        This function creates a HotSpare
+        """
+        if self.number_of_disk > 2:
+            cmd = "echo -e 'YES\nNO' | ./sas3ircu %d hotspare %s" \
+                 % (self.controller, self.drives[2])
+        if process.system(cmd, ignore_status=True, shell=True) != 0:
+            self.fail("Failed to set hotspare drive")
```
Comment (narasimhan-v) [inline io/disk/lsi3008.py]: This command should be inside `if`

Comment (PraveenPenguin) [inline io/disk/lsi3008.py]: a standard practice to include at the end
if name == "main":
main()

File: io/disk/lsi3008.py.data/Readme [hunk 4]
```diff
@@ -0,0 +1,9 @@
+This scripts runs basic sanity test for the LSI-3008 adapter.
+
+Values to be passed in YMAL file:
+
+Controller ID of the adapter, for which these tests will be executed.
+
+Disks bay ID. i.e., both enclosure ID and slot ID, as shown below.
+en:sn, where en --> enclosure ID, sn --> slot ID
+
```
Comment (PraveenPenguin) [inline io/disk/lsi3008.py.data/Readme]: please remove this white space

File: io/disk/lsi3008.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    """
+
+    def setUp(self):
+
+        """
+        Lists all available Avago adapters (does not need ctlr #>
+        """
+
+        self.controller = int(self.params.get('controller', default='0'))
+        self.disk = self.params.get('disk', default='').strip(" ").split(" ")
```
Comment (narasimhan-v) [inline io/disk/lsi3008.py]: Handle the case when the parameter `disk` is empty

File: io/disk/lsi3008.py [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+        """
+        This function set's a drive to offline
+        """
+
+        cmd = "./sas3ircu %d setoffline %s" \
+              % (self.controller, self.drives[0])
+        if process.system(cmd, ignore_status=True, shell=True) != 0:
+            self.fail("Failed to set drive to offline")
+        time.sleep(60)
```
Comment (PraveenPenguin) [inline io/disk/lsi3008.py]: can you please help me to understand why this sleep is required ?

Comment (vrbagalkote) [inline io/disk/lsi3008.py]: Hi @PraveenPenguin ,

I just want raid array to be in degraded mode for some time before coming back to optimal state.

Thanks,
Venkat.

File: io/disk/lsi3008.py (hunk 1 above)
Comment (vrbagalkote) [inline io/disk/lsi3008.py]: Taken care.

File: io/disk/lsi3008.py (hunk 2 above)
Comment (vrbagalkote) [inline io/disk/lsi3008.py]: taken care.

File: io/disk/lsi3008.py (hunk 3 above)
Comment (vrbagalkote) [inline io/disk/lsi3008.py]: Done.

File: io/disk/lsi3008.py (hunk 5 above)
Comment (vrbagalkote) [inline io/disk/lsi3008.py]: Done

File: io/disk/lsi3008.py (hunk 3 above)
Comment (vrbagalkote) [inline io/disk/lsi3008.py]: Done

File: io/disk/lsi3008.py.data/Readme (hunk 4 above)
Comment (vrbagalkote) [inline io/disk/lsi3008.py.data/Readme]: Done

File: io/disk/lsi3008.py (hunk 6 above)
Comment (PraveenPenguin) [inline io/disk/lsi3008.py]: @vrbagalkote sure thanks

**Comment by vrbagalkote** [comment]
Created a new pull request, So I am closing this.

https://github.com/avocado-framework-tests/avocado-misc-tests/pull/175

**Pull request closed without merging**